app/src/main/java/org/simple/clinic/encounter/EncounterRepository.kt

```diff
@@ -17,13 +17,13 @@ import javax.inject.Inject
 
 class EncounterRepository @Inject constructor(
     private val database: AppDatabase
-) : SynceableRepository<Encounter, EncounterPayload> {
+) : SynceableRepository<ObservationsForEncounter, EncounterPayload> {
 
-  override fun save(records: List<Encounter>): Completable {
-    return Completable.fromAction { database.encountersDao().save(encounters = records) }
+  override fun save(records: List<ObservationsForEncounter>): Completable {
+    return saveMergedEncounters(records)
```

Comment:
I looked at the `saveMergedEncounters` method implementation, and I saw the following code block:

```kotlin
with(database) {
  openHelper.writableDatabase.inTransaction {
    bloodPressureDao().save(bloodPressures)
    encountersDao().save(encounters)
  }
}
```

The `SupportSQLiteDatabase#inTransaction` extension method was written to be used in migrations and the migration tests, since we do not have access to the `AppDatabase` in those classes.

If you have access to the `AppDatabase`, Room already provides a `runInTransaction` method which is more convenient instead of accessing the internal open helper.

```kotlin
with(database) {
  runInTransaction {
    bloodPressureDao().save(bloodPressures)
    encountersDao().save(encounters)
  }
}
```

Would you like to switch to using this method instead?

Reply:
If we do, we should probably also add a comment on the `SupportSQLiteDatabase#inTransaction` method that clarifies where it is meant to be used and pointing users to the Room method as an alternative.